### PR TITLE
Remove a hard assert that no longer holds due to mirror queries.

### PR DESCRIPTION
**Pull request by wu-hui**

### CODE REVIEW

File: firebase-firestore/src/main/java/com/google/firebase/firestore/core/SyncEngine.java [hunk 1]
```diff
@@ -220,9 +220,6 @@ private ViewSnapshot initializeViewAndComputeSnapshot(Query query, int targetId)
     View view = new View(query, queryResult.getRemoteKeys());
     View.DocumentChanges viewDocChanges = view.computeDocChanges(queryResult.getDocuments());
     ViewChange viewChange = view.applyChanges(viewDocChanges, synthesizedCurrentChange);
-    hardAssert(
```
Comment (schmidt-sebastian) [inline firebase-firestore/src/main/java/com/google/firebase/firestore/core/SyncEngine.java:223]: I believe the reason that this assert was here is that this method does not call `updateTrackedLimboDocuments`. 

So, we either need to convince ourselves that any limbo document encountered here will already be in limbo thanks to the originating query (and document this assumption) or call `updateTrackedLimboDocuments`.

Comment (wu-hui) [reply firebase-firestore/src/main/java/com/google/firebase/firestore/core/SyncEngine.java:223]: Added a call to updateTrackedLimboDocuments

**Comment by schmidt-sebastian** [comment]
LGTM. Please remember to port to iOS and Web.

**Pull request merged**